src/core/Form/RadioButton/RadioButtonGroup.tsx

```diff
@@ -1,23 +1,152 @@
+import { RadioButtonProps } from './RadioButton';
 import { baseStyles } from './RadioButtonGroup.baseStyles';
 import { withSuomifiDefaultProps } from '../../theme/utils';
+import { AutoId } from '../../../utils/AutoId';
 import classnames from 'classnames';
 
 const baseClassName = 'fi-radio-button-group';
+const radioButtonGroupClassNames = {
+  container: `${baseClassName}_container`,
+  label: `${baseClassName}_label`,
+  hintText: `${baseClassName}_hintText`,
+};
 
-export interface RadioButtonGroupProps
-  extends CompRadioButtonGroupProps,
-    TokensProp {}
+type Label = 'hidden' | 'visible';
+
+export interface RadioButtonGroupProps extends TokensProp {
+  /** Custom classname to extend or customize */ className?: string;
+  /** RadioButton or ReactNode */
+  children: Array<React.ReactElement<RadioButtonProps> | ReactNode>;
+  /** Hint text to be displayed under the label. */
+  hintText?: string;
+  /** Label for the group */
+  label: string;
+  /** Hide or show label. Label element is always present, but can be visually hidden.
+   * @default visible
+   */
+  labelMode?: Label;
+  /**
+   * Unique id
+   * If no id is specified, one will be generated
+   */
+  id?: string;
+  /** Name for the group; this will be set to RadioButtons. Overrides RadioButton props name. */
+  name: string;
+  /** Controlled value of the selected RadioButton in the group. Overrides RadioButton selected values. */
+  value?: string;
+  /** Value of the RadioButton selected by default */
+  defaultValue?: string;
+  /** Callback for RadioButtonGroup selected changes. */
+  onChange?: (value: string) => void;
+}
+
+export interface RadioButtonGroupProviderState {
+  onRadioButtonChange: (value: string) => void;
+  name?: string;
+  selectedValue?: string;
+}
+
+const defaultProviderValue: RadioButtonGroupProviderState = {
+  onRadioButtonChange: () => null,
+};
+
+const { Provider, Consumer: RadioButtonGroupConsumer } = React.createContext(
+  defaultProviderValue,
+);
+
+export interface RadioButtonGroupState {
+  selectedValue?: string;
+}
+
+export class BaseRadioButtonGroup extends Component<RadioButtonGroupProps> {
+  state: RadioButtonGroupState = {
+    selectedValue: this.props.value || this.props.defaultValue,
+  };
+
+  static getDerivedStateFromProps(
+    nextProps: RadioButtonGroupProps,
+    prevState: RadioButtonGroupState,
+  ) {
+    const { value } = nextProps;
+    if (value !== undefined && value !== prevState.selectedValue) {
+      return { selectedValue: value };
+    }
+    return null;
+  }
+
+  handleRadioButtonChange = (value: string) => {
+    if (!!this.props.onChange) {
+      this.props.onChange(value);
+    }
+    if (!('value' in this.props)) {
+      this.setState({ selectedValue: value });
+    }
+  };
+
+  render() {
+    const {
+      children,
+      className,
+      label,
+      labelMode,
+      hintText,
+      id,
+      name,
+      defaultValue: dismissDefaultValue,
```

Comment:
As discussed earlier in separate issue, should we get rid of this `dismissDefaultValue` and just leave it as `defaultValue` ?

Reply:
'discard' would be better wording, but I'm kind of torn between leaving it as is and not explicitly stating that this value is not used here.

Reply:
Removed the rename completely.

Reply:
Finally fixed. Mixed this up with similar issue in RadioButton.